txp/Pos/Txp/Toil/Types.hs

```diff
@@ -124,18 +127,41 @@ instance Default MemPool where
         }
 
 ----------------------------------------------------------------------------
--- ToilModifier
+-- UtxoModifier, UndoMap and AddrCoinsMap
 ----------------------------------------------------------------------------
 
 type UtxoModifier = MM.MapModifier TxIn TxOutAux
 type UndoMap = HashMap TxId TxUndo
-
-fromUtxo :: Utxo -> UtxoModifier
-fromUtxo = foldr (uncurry MM.insert) mempty . M.toList
+type AddrCoinMap = HashMap Address Coin
+
+utxoToModifier :: Utxo -> UtxoModifier
+utxoToModifier = foldr (uncurry MM.insert) mempty . M.toList
```

Comment:
Just genuine curiosity more than anything else: What's dictate the choice of `foldr` here? (I know this is not subject of review as the previous implementation was using that as well).

Reply:
`foldr` has type ` (a -> b -> b) -> b -> t a -> b` and I don't have to do `flip $ uncurry MM.insert`

Reply:
Gotcha! It would be interesting to see the performance difference (if any) when using `foldl'` instead, as here we have a finite input (the one we are building out of the `Map`) and we don't need to short-circuit to find a result. But nothing that should stop the PR from being merged 😉